Automattic/newspack-plugin
# fix(ras): add script for fixing duplicate Mailchimp merge fields

### All Submissions:






### Changes proposed in this Pull Request:



This PR adds a CLI script related to but slightly different from the script to delete specific merge fields #3623. The new script identifies duplicate instances of merge fields with the same name in the connected Mailchimp account, then consolidates data in all instances of the field for all members in the audience into the single oldest instance of the field, then deletes all other instances of the field.

### How to test the changes in this Pull Request:

1. Check out this branch.
2. In your connected Mailchimp account, go to **Audience > All contacts > Settings > Audience fields & \*MERGE\* tags** and add at least one new merge field with the same name as an existing Newspack meta field. You could add multiple duplicates of the same field and/or duplicates of different Newspack fields, if your audience has enough available merge fields.
3. In at least one contact in the audience, manually edit the contact data and add some data to the newer duplicate field(s).
4. Run `wp newspack mailchimp merge-fields fix-duplicates --dry-run` and confirm that it reports the duplicate fields but doesn't make any changes to the Mailchimp data or fields:

```
Fixing duplicate merge fields in audience <audience ID>... (DRY RUN MODE)

Found 1 duplicate merge field(s) for NP_Registration Page

Success: Would consolidate data in duplicate fields for audience <audience ID>
		Deleting merge field MMERGE26 (26) for NP_Registration Page
		DRY RUN: would have deleted merge field MMERGE26 (NP_Registration Page) in audience <audience ID>.

Success: Would delete duplicate fields for audience <audience ID>: NP_Registration Page
```

5. Run the command without the `--dry-run` flag and confirm in your Mailchimp account that it a.) consolidates all of the contact data you added to the duplicate fields into the oldest instance of the field, and b.) deletes all other instances of the field.

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208959720314117
